src/xmlwriter.rs

```diff
@@ -44,17 +45,15 @@ impl XMLWriter {
 
     // Reset the memory buffer, usually between saves.
     pub(crate) fn reset(&mut self) {
-        let buf: Vec<u8> = Vec::with_capacity(2048);
-        self.xmlfile = Cursor::new(buf);
+        self.xmlfile.get_mut().clear();
+        self.xmlfile.set_position(0);
     }
 
     // Write an XML file declaration.
     pub(crate) fn xml_declaration(&mut self) {
-        writeln!(
-            &mut self.xmlfile,
-            r#"<?xml version="1.0" encoding="UTF-8" standalone="yes"?>"#
-        )
-        .expect("Couldn't write to file");
+        self.xmlfile
+            .write_all(b"<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n")
```

Comment:
Any particular reason for using `write_all()` instead of `write()` here and below?

Reply:
`write_all()` bypasses the formatter. In general, anything from the `fmt` module is not free. Not this specific one but I saw some other `write_fmt`s on flamegraph. Any time you can is better to bypass the formatter.